cern_access/indico_cern_access/plugin.py

```diff
@@ -153,6 +157,16 @@ def _get_access_status(self, regform, registration, header, **kwargs):
                                           registration=registration,
                                           header=header)
 
+    def _get_personal_data_form(self, event, regform, management, registration=None, **kwargs):
+        if management or registration is not None:
+            return
+
+        form = RegistrationFormPersonalDataForm()
```

Comment:
If you use `form = g.get('personal_data_form') or RegistrationFormPersonalDataForm()` here you should get proper form error display without having to flash them manually